hazelcast/src/main/java/com/hazelcast/map/IMap.java

```diff
@@ -1762,6 +1764,82 @@ void set(@Nonnull K key, @Nonnull V value,
              long ttl, @Nonnull TimeUnit ttlUnit,
              long maxIdle, @Nonnull TimeUnit maxIdleUnit);
 
+    /**
+     * Copies all of the mappings from the specified map to this map without loading
+     * non-existing elements from map store (which is more efficient than {@code putAll()}).
+     * Do not concurrently mutate the given map.
+     * <p>
+     * This method breaks the contract of EntryListener.
+     * EntryEvent of all the updated entries will have null oldValue even if they exist previously.
+     * <p>
+     * No atomicity guarantees are given. It could be that in case of failure
+     * some of the key/value-pairs get written, while others are not.
+     *
+     * <p><b>Interactions with the map store</b>
+     * <p>
+     * If write-through persistence mode is configured,
+     * {@link MapStore#store(Object, Object)} is invoked for each element
+     * before the element is added in memory, which may come at a
+     * significant performance cost. Exceptions thrown by store fail the
+     * operation and are propagated to the caller. The elements which
+     * were added before the exception was thrown will remain in the map,
+     * the rest will not be added.
+     * <p>
+     * If write-behind persistence mode is configured with
+     * write-coalescing turned off,
+     * {@link com.hazelcast.map.ReachedMaxSizeException} may be thrown
+     * if the write-behind queue has reached its per-node maximum
+     * capacity.
+     */
+    void setAll(@Nonnull Map<? extends K, ? extends V> map);
+
+    /**
+     * Asynchronously copies all of the mappings from the specified map to this map
+     * without loading non-existing elements from map store. This version doesn't
```

Comment:
`This version doesn't support batching.` -> it's an implementation detail, not relevant to the contract.
`Don't mutate the given map until the future completes.` -> why? It's a pretty bad requirement, most multi-threaded applications won't be able to adhere to this rule.